src/app/store/shopping-list/shopping-list.effects.ts

```diff
@@ -149,6 +149,32 @@ export class shoppingListEffects {
 		);
 	});
 
+	updateIngredientsFromRecipe = createEffect(() => {
+		return this.actions$.pipe(
+			ofType(shoppingListActions.updateIngredient),
+			concatLatestFrom(() =>
+				this.store.select(ShoppingListSelectors.selectAll)
+			),
+			map(([action, ingredientInShoppingList]: [any, IngredientsInStore[]]) => {
```

Comment:
You don't need to add ` [any, IngredientsInStore[]]`, I think you should have types already from store

Reply:
I fix it